### PR TITLE
mlflow recipe

### PR DESCRIPTION
**Pull request by jaroslawk**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jaroslawk** [comment]
@conda-forge/core it looks like the linter did not picked up my recipe. Anyone can help here?

**Comment by dbast** [comment]
Maybe rename the `meta.yml` to `meta.yaml`.

**Comment by jaroslawk** [comment]
@dbast silly mistake. Thx! What should I do when there are additional dependencies mlfow requires? just attempt to add them as additional recipes, right?

**Comment by dbast** [comment]
Sure, if some dependencies are missing, those then require separate PRs.

**Comment by jaroslawk** [comment]
ok Thank you!

**Comment by dbast** [comment]
@jaroslawk Two hints:
* Compare your meta.yaml for more inspiration with the output of: `conda-skeleton pypi mlflow`
* Seems, like the pypi tarball already contains a LICENSE.txt you can use via the meta.yaml without including the file in the recipe.

File: recipes/mlflow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: c2c76d072f4735a229333095506acb3d6ec0305656ee7fc468e9b60a0e8d7b8a
+
+build:
+  number: 0
+  entry_points:
+    - mlflow=mlflow.cli:cli
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+  skip: True  # [not win]
```
Comment (xhochy) [inline recipes/mlflow/meta.yaml]: `mlflow` doesn't work on windows due to the missing set of dependencies. Thus this condition must be the other way around.

```suggestion
  skip: True  # [win]
```

**Comment by jaroslawk** [comment]
@xhochy I wonder how can I solve the issue with missing docker and mleap dependency. It is acceptable to package it with mlflow?

**Comment by xhochy** [comment]
No, they should be separate conda-forge packages. I'm already taking a look at `docker(-py)`

**Comment by jaroslawk** [comment]
@xhochy ok perfect. Thank you!

File: recipes/mlflow/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+requirements:
+  build:
+    - python
+    - setuptools
+    - pip
+  run:
+    - click >=7.0
+    - cloudpickle ==0.6.1
+    - databricks-cli >=0.8.0
+    - docker >=3.6.0
```
Comment (xhochy) [inline recipes/mlflow/meta.yaml]: Already exists as `docker-py` 😂 

```suggestion
    - docker-py >=3.6.0
```

Comment (jaroslawk) [reply recipes/mlflow/meta.yaml]: there is some mess with this dependency. 
in pypi there is docker and docker-py
docker-py is old one and apparently, they did not update their conda recipe

Comment (xhochy) [reply recipes/mlflow/meta.yaml]: AFAICS `docker-py` on conda-forge is what `docker` is on PyPI, it is also uptodate with the `docker` pypi package version. So using this should be the right thing.

**Comment by jaroslawk** [comment]
@xhochy ok I fill convinced about the docker-py/docker mess.. ;-)

**Comment by xhochy** [comment]
`mleap` is now also packaged and uploaded, just `databricks-cli` is missing.

**Comment by jaroslawk** [comment]
@xhochy seen it. Thank you! This was a huge help! 

It looks like databricks-cli PR is mostly done I will finish this one soon.

File: recipes/mlflow/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    - mlflow.tracking
+    - mlflow.utils
+  commands:
+    - mlflow --help
+
+about:
+  home: https://mlflow.org/
+  license: Apache 2.0
+  license_family: APACHE
+  license_file:
```
Comment (dbast) [inline recipes/mlflow/meta.yaml]: either the pypi tarball contains a LICENSE file that can be referenced here or the LICENSE can be packaged as it was done in e.g. the databricks-cli package.

File: recipes/mlflow/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+test:
+  imports:
+    - mlflow
+    - mlflow.azureml
+    - mlflow.entities
+    - mlflow.models
+    - mlflow.projects
+    - mlflow.protos
+    - mlflow.pyfunc
+    - mlflow.pytorch
```
Comment (dbast) [inline recipes/mlflow/meta.yaml]: This seems to require pytorch, which looks like to be optional as also not listed as requirement in the setup.py (it is not available in conda-forge). Maybe deactivate that line for now to see if the recipe builds successful.

File: recipes/mlflow/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    - python-dateutil
+    - pyyaml
+    - querystring_parser
+    - requests >=2.17.3
+    - scikit-learn
+    - scipy
+    - simplejson
+    - six >=1.10.0
+    - sqlparse
+    - boto3 >=1.7.12
```
Comment (dbast) [inline recipes/mlflow/meta.yaml]: maybe follow alphabetical ordering here?

File: recipes/mlflow/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
+{% set name = "mlflow" %}
+{% set version = "0.9.0" %}
```
Comment (dbast) [inline recipes/mlflow/meta.yaml]: There is now 0.9.0.1

**Comment by jaroslawk** [comment]
@andrewmchen please can you confirm you agree to be added as a supporter to mlflow conda forge?

**Comment by andrewmchen** [comment]
@jaroslawk, yes. Thank you for doing this for the MLflow project.

**Comment by dbast** [comment]
@jaroslawk You can keep noarch without skip. The noarch recipe will be anyway built on linux after feedstock conversion. It won't be installable on windows due to non availability of gunicorn, which is fine, see also https://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages

> noarch: python can also work with recipes that would work on a given platform except that we don’t have one of its dependencies available. If this is the case, when the install runs conda will fail with an error message describing which dependency couldn’t be found. If the dependency is later made available, your package will start working without having to change anything. Note though that since noarch packages are built on Linux, currently the package must be installable on Linux.

File: recipes/mlflow/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    - mlflow.tracking
+    - mlflow.utils
+  commands:
+    - mlflow --help
+
+about:
+  home: https://mlflow.org/
+  license: Apache 2.0
+  license_family: APACHE
+  license_file: LICENSE
```
Comment (dbast) [inline recipes/mlflow/meta.yaml]: This should be `license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'` ... according to https://conda-forge.readthedocs.io/en/latest/meta.html#packaging-the-license-manually

**Comment by jaroslawk** [comment]
@dbast when I place noarch there are check's failing for the win. Should I ignore it?
How can this be fixed?

**Comment by dbast** [comment]
Ignoring afaik is fine here... it is a noarch recipe and only a dependency is missing on a specific arch... see the cited text.

**Comment by dbast** [comment]
Looks like this is ready for review/merge :)

**Comment by dbast** [comment]
@conda-forge/help-python ^

**Comment by jaroslawk** [comment]
@conda-forge/help-python this PR should be good to go. Can we merge it?

**Comment by xhochy** [comment]
@conda-forge/staged-recipes general ping here. The package is looking fine, the windows failure is simply due to the package not supporting Windows (it is noarch and a dependency is missing on Win). So this is ready to be merged.

File: recipes/mlflow/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    - mlflow.tracking
+    - mlflow.utils
+  commands:
+    - mlflow --help
+
+about:
+  home: https://mlflow.org/
+  license: Apache 2.0
+  license_family: APACHE
+  license_file: '{{ environ["RECIPE_DIR"] }}/LICENSE'
```
Comment (chrisburr) [inline recipes/mlflow/meta.yaml]: There is no need for the `RECIPE_DIR` here:
```suggestion
  license_file: LICENSE
```
(I'll merge once this is done and the CI passes, ping me if needed)

**Pull request merged**